src/GitVersionCore/Core/Abstractions/IRepositoryStore.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace GitVersion.Common
 {
-    public interface IRepositoryMetadataProvider
+    public interface IRepositoryStore
```

Comment:
I would keep them in store for now. The reason is I want to have the GitRepository and it's collections as simple as possible, just a proxy for the git implementation (either libgit2harp, or any other implementation we might have). That will reduce the API surface we need to implement for a new git implementation (basically the git implementation would expose the Git Objects that can then we queried by the level on top - store).

![image](https://user-images.githubusercontent.com/1760506/105986264-f9614c00-609c-11eb-99c1-83f297aeb496.png)

Reply:
But the `ReachableFrom()` method would need to know about the underlying implementation regardless of where it's placed, no?

Reply:
Actually no. The `GetCommitsReacheableFrom` method uses the IGitRepository.Commits that is abstract. It does not need to know about the underlying implementation, it just knows there is collection it can perform queries. The implementation takes care to transform it to internal querying that is either in memory or actually quering the git database

Reply:
Doesn't `GetCommitsReacheableFrom` use `CommitFilter` which is defined in LibGit2Sharp? I assume the implementation is going to use a different query mechanism if LibGit2Sharp is not the underlying abstraction?

Reply:
Well the `CommitFilter` it's defined in the Core, and is based on the `LibGit2Sharp` version, but it could evolve later into a more abstract mechanism. But as it is now it's not coupled to `LibGit2Sharp`,  it just gets converted into `CommitFilter` from `LibGit2Sharp`

Reply:
I imagine an implementation of `CommitFilter` for another underlying abstraction is going to be difficult. If we don't expose `CommitFilter` publicly, I think it's better to delete our abstraction of it and just leave it as an implementation detail.

Reply:
It is public - https://github.com/GitTools/GitVersion/blob/master/src/GitVersionCore/Git/CommitFilter.cs

Reply:
Yes, but I believe it shouldn't be since implementing `CommitFilter` for underlying abstractions other than LibGit2Sharp seems like much work.

Reply:
I think I will keep it like this in the PR, so that we can have more time to think.